src/plugins/dashboard/public/dashboard_app/top_nav/editor_menu.tsx

```diff
@@ -41,6 +46,97 @@ interface UnwrappedEmbeddableFactory {
   isEditable: boolean;
 }
 
+export type GetEmbeddableFactoryMenuItem = ReturnType<typeof getEmbeddableFactoryMenuItemProvider>;
+
+export const getEmbeddableFactoryMenuItemProvider =
+  (api: PresentationContainer, closePopover: () => void) => (factory: EmbeddableFactory) => {
+    const icon = factory?.getIconType ? factory.getIconType() : 'empty';
+
+    const toolTipContent = factory?.getDescription ? factory.getDescription() : undefined;
+
+    return {
+      name: factory.getDisplayName(),
+      icon,
+      toolTipContent,
+      onClick: async () => {
+        closePopover();
+        api.addNewPanel({ panelType: factory.type }, true);
+      },
+      'data-test-subj': `createNew-${factory.type}`,
+    };
+  };
+
+export const mergeGroupedItemsProvider =
```

Comment:
Is there a reason these need to be curried functions? Could it be simpler with all arguments taken in one function call?

Reply:
Just to separate the dependencies of the function and the input arguments. I don't have strong preferences here, so whatever pattern the team prefer